src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -150,10 +157,39 @@ private void OnGetSuggestionsCommandExecute()
                 _suggestionsDropdownTelemetry.DropdownButtonClickTimes += 1;
             }
 
-            _getSuggestionsTask = _smartRenameSession.GetSuggestionsAsync(_cancellationTokenSource.Token).CompletesAsyncOperation(listenerToken);
+            _getSuggestionsTask = GetSuggestionsTaskAsync(_cancellationTokenSource.Token).CompletesAsyncOperation(listenerToken);
         }
     }
 
+    private async Task GetSuggestionsTaskAsync(CancellationToken cancellationToken)
+    {
+        var document = this.BaseViewModel.Session.TriggerDocument;
+        var editorRenameService = document.GetRequiredLanguageService<IEditorInlineRenameService>();
+        var context = await editorRenameService.GetRenameContextAsync(this.BaseViewModel.Session, cancellationToken)
+            .ConfigureAwait(true);
+
+        var symbolService = document.GetLanguageService<IGoToDefinitionSymbolService>();
+        if (symbolService is not null)
+        {
+            var textSpan = this.BaseViewModel.Session.RenameInfo.TriggerSpan;
+            var (symbol, _, _) = await symbolService.GetSymbolProjectAndBoundSpanAsync(
```

Comment:
I feel this needs to be moved to `LanguageService` (e.g. the derived type of AbstractEditorInlineRenameService)
`SmartRenameViewModel` would be created for other language (like xaml) and they might don't have the concept of `Symbol`

Reply:
Maybe we can move this whole method to the language service.